clippy_lints/src/operators/decimal_bitwise_operands.rs

```diff
@@ -0,0 +1,65 @@
+use clippy_utils::diagnostics::span_lint_and_help;
+use clippy_utils::source::SpanRangeExt;
+use rustc_ast::LitKind;
+use rustc_data_structures::packed::Pu128;
+use rustc_hir::{AssignOpKind, BinOpKind, Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_span::Span;
+use rustc_span::source_map::Spanned;
+
+use super::DECIMAL_BITWISE_OPERANDS;
+
+pub(super) fn check_binary<'tcx>(cx: &LateContext<'tcx>, op: BinOpKind, left: &'tcx Expr<'_>, right: &'tcx Expr<'_>) {
```

Comment:
After moving this lint to the operators category (https://github.com/rust-lang/rust-clippy/pull/15215/commits/a6c0cfd6071a0968707d3221d24f60a9f95eb896), I decided to split the `check` function into `check_binary` and `check_assign`. This makes it more efficient, because it avoids redundant pattern matching that is already performed in `src/operators/mod.rs`.